### ISSUE TITLE
XLA: specify num_partitions in xla_cpu_runtime_ParallelForkJoin

### ISSUE THREAD

**Issue by SerenaC94** [tensorflow/tensorflow#27836]
I am interested in the LLVM IR obtained by launching a Tensorflow test program with the `--xla_dump_ir_to` flag. In particular, I would like XLA to create a certain number of copies of a function that can run in parallel on multiple CPUs. 
(I know this sounds like a support question rather than an issue, but nobody could answer me on StackOverflow until now, so I figured that maybe what I am looking for is a feature that still does not exist)

I have a test program that includes these lines
`with tf.device("device:XLA_CPU:0"):`
`       y = tf.nn.conv2d(X, weights_tensor, strides=[1, 1, 1, 1], padding='SAME')`
`    sess.run(tf.global_variables_initializer())`

And in the output `.ll` files XLA generates a call to
`call void @__xla_cpu_runtime_ParallelForkJoin(i8* %6, i8* %run_options, i8** null, i8** %buffer_table, i64* %prof_counters, i32 56, i64* getelementptr inbounds ([224 x i64], [224 x i64]* @parallel_convolution_parallel_dimension_partitions, i32 0, i32 0), i32 2, i8* bitcast (void (i8*, i8*, i8**, i8**, i64*, i64*)* @parallel_convolution to i8*))`
which is a good starting point.

But now I want to specify how many copies of that function I want. What I have tried up to now is to use these options
`config = tf.ConfigProto()`
`config.device_count={"CPU": 5}`
`config.intra_op_parallelism_threads=5`
`config.inter_op_parallelism_threads=5`
and they still lead to the same call, with the `num_partitions` argument set to 56 (number of CPUs on my machine).

Moreover, a different test program with
`with tf.device("device:XLA_CPU:0"):`
`                y=tf.layers.max_pooling2d(inputs=x, pool_size=[3, 3], strides=3)`
`        sess.run(y,{x: reshaped_input_array})`
does not generate any call to `xla_cpu_runtime_ParallelForkJoin` at all.

Is it possible to force XLA in any way to produce parallel copies of a function, and to specify how many copies?

### System information
- OS Platform and Distribution: Linux 4.19.0-4-amd64 #1 SMP Debian 4.19.28-2 (2019-03-15)
- TensorFlow installed from: source
- TensorFlow version: 1.13.1
- Python version: 2.7.16
- Bazel version: 0.21.0
- GCC version: 8.3.0

**Comment by ymodak** [comment]
@jlebar

**Comment by jlebar** [comment]
> Is it possible to force XLA in any way to produce parallel copies of a function, and to specify how many copies?

No, exactly how XLA generates LLVM IR for a particular op is an implementation detail.

If you're interested in contributing patches to add this feature, xla-dev@googlegroups.com would be a place to start the discussion.
